WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -406,20 +417,32 @@ private extension DashboardViewModel {
     /// this should be updated to general app settings.
     ///
     @MainActor
-    func updateDashboardCards(canShowOnboarding: Bool, canShowBlaze: Bool) async {
+    func updateDashboardCards(canShowOnboarding: Bool,
+                              canShowBlaze: Bool,
+                              canShowAnalytics: Bool) async {
         dashboardCards = await {
             if let stored = await loadDashboardCards() {
                 return stored
             } else {
```

Comment:
I found an edge case that I have dashboard cards saved locally, so the conditional checks are bypassed. Should we update the loaded card with the checks, save the new settings locally and return the updated settings?

Reply:
I also run into this situation when initially starting from trunk then working on this branch. Do you think this happens in real life? In any case let's continue discussing this here, then I'll work on a separate PR if a new change is needed.

Reply:
I ended up adding the functionality to take into account and update local changes in the next PR:

https://github.com/woocommerce/woocommerce-ios/pull/12545/